### ISSUE TITLE
Docker support

### ISSUE THREAD

**Issue by AndreiMaz** [nopSolutions/nopCommerce#3113]
see https://hub.docker.com/ (https://docs.docker.com/docker-hub/)
 and https://docs.docker.com/#/typical-docker-platform-workflow

also check https://www.nopcommerce.com/boards/t/51242/will-nop-41-support-docker-clustering-.aspx

**Comment by pccruiser** [comment]
Some official support would be great. I'm struggling to even get the Visual Studio tools to work here - hoping that 4.1 will fix that since it can be full on .net core and we can drop net461 completely. Having to move to a manual docker image creation workflow.

Having clustering and orchestration will be very nice. Maybe less dependency on file system with data settings in json and utilize Azure Key Vault?

Could make that as part of the setup - put in config settings for keyvault - use those in session to auth in, grab DB settings and then move on?

**Comment by VR-Architect** [comment]
+1 for an official docker version in GitHub.
+2 with instructions for nginx docker pre-configured for nopCommerce

**Comment by pccruiser** [comment]
Good news! I was able to pull the 4.1 beta down and confirm my suspicion - having a true full as.net core project (not a weird mix of core targeting net461) and Docker tools worked out of the box on the new solution! Plane landed so I had to close laptop and haven't actually run the container yet but that is a big step.

Still - having an official Docker image to use as a Base is a must. This is the future of development and deployment.
